src/data/store.rs

```diff
@@ -149,40 +150,41 @@ impl Store {
         prompt: String,
         regenerate_from: Option<usize>,
     ) {
-        let entity = self
+        let entity_id = self
             .chats
             .get_current_chat()
             .and_then(|c| c.borrow().associated_entity.clone());
 
-        match entity {
-            Some(ChatEntity::Agent(agent)) => {
-                self.send_agent_message(agent, prompt, regenerate_from);
-            }
-            _ => {
-                self.send_chat_message(prompt, regenerate_from);
-            }
+        if let Some(entity_id) = entity_id {
+            self.send_entity_message(&entity_id, prompt, regenerate_from);
         }
     }
 
-    pub fn send_chat_message(&mut self, prompt: String, regenerate_from: Option<usize>) {
+    pub fn send_entity_message(
```

Comment:
Some refactoring here. This method handles what `send_chat_message` and `send_agent_message` handled, but allowing to specify a specific model, and also covering the last part of the `send_message_to_current_entity` method.